### ISSUE TITLE
IMAP4 protocol error: program error: <class 'TypeError'> - can't use a bytes pattern on a string-like object

### ISSUE THREAD

**Issue by ZaxonXP** [OfflineIMAP/offlineimap#607]
#### General informations

- system/distribution (with version): Debian GNU/Linux 8.11 (jessie)
- offlineimap version (`offlineimap -V`): offlineimap v7.2.3, imaplib2 v2.57 (bundled), Python v3.4.2, OpenSSL 1.0.1t  3 May 2016
- Python version: Python 2.7.9
- server name or domain: 
- CLI options: 

#### Configuration file offlineimaprc

```
[general]
accounts = GMail
maxsyncaccounts = 1
ui = basic

[Account GMail]
localrepository = Local
remoterepository = Remote

[Repository Local]
type = Maildir
localfolders = ~/Mail/GMail/

[Repository Remote]
type = Gmail
remoteuser = <private>
remotepass = <private>
folderfilter = lambda folder: folder not in ['[Gmail]/All Mail','[Gmail]/Important', '[Gmail]/Spam', '[Gmail]/Bin']

sslcacertfile = /etc/ssl/certs/ca-certificates.crt

```

#### pythonfile (if any)

```
REMOVE PRIVATE DATA.
```

#### Logs, error

```
OfflineIMAP 7.2.3
  Licensed under the GNU GPL v2 or any later version (with an OpenSSL exception)
imaplib2 v2.57 (bundled), Python v3.4.2, OpenSSL 1.0.1t  3 May 2016
*** Processing account GMail
Establishing connection to imap.gmail.com:993 (Remote)
ERROR: While attempting to sync account 'GMail'
  IMAP4 protocol error: program error: <class 'TypeError'> - can't use a bytes pattern on a string-like object
*** Finished account 'GMail' in 0:00
ERROR: Exceptions occurred during the run!
ERROR: While attempting to sync account 'GMail'
  IMAP4 protocol error: program error: <class 'TypeError'> - can't use a bytes pattern on a string-like object

Traceback:
  File "/usr/local/lib/python3.4/dist-packages/offlineimap/accounts.py", line 283, in syncrunner
    self.__sync()
  File "/usr/local/lib/python3.4/dist-packages/offlineimap/accounts.py", line 359, in __sync
    remoterepos.getfolders()
  File "/usr/local/lib/python3.4/dist-packages/offlineimap/repository/IMAP.py", line 452, in getfolders
    imapobj = self.imapserver.acquireconnection()
  File "/usr/local/lib/python3.4/dist-packages/offlineimap/imapserver.py", line 572, in acquireconnection
    af=self.af,
  File "/usr/local/lib/python3.4/dist-packages/offlineimap/imaplibutil.py", line 194, in __init__
    super(WrappedIMAP4_SSL, self).__init__(*args, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/offlineimap/bundled_imaplib2.py", line 2183, in __init__
    IMAP4.__init__(self, host, port, debug, debug_file, identifier, timeout, debug_buf_lvl)
  File "/usr/local/lib/python3.4/dist-packages/offlineimap/bundled_imaplib2.py", line 400, in __init__
    self.welcome = self._request_push(name='welcome', tag='continuation').get_response('IMAP4 protocol error: %s')[1]
  File "/usr/local/lib/python3.4/dist-packages/offlineimap/bundled_imaplib2.py", line 201, in get_response
    raise typ(exc_fmt % str(val))


```

#### Steps to reproduce the error

just run `offlineimap`

**Comment by nicolas33** [comment]
`imaplib2 v2.57 (bundled), Python v3.4.2,`

offlineimap won't work with py3.

**Comment by ZaxonXP** [comment]
So, how can I tell offlineimap to use python 2.x and not 3.x?

**Comment by nicolas33** [comment]
I have no idea why it doesn't work for you. How do you start offlineimap?
What's the output of:
- `python2 --version`
- `env python2 --version`
Did you start pyvenv or similar at some point in time?

**Comment by nicolas33** [comment]
Notice: the offlineimap executable script is a small file. Check the shebang with:
`head $(which offlineimap)`
This might help too: `whereis offlineimap`

**Comment by ZaxonXP** [comment]
Looks like I had system package and one installed with pip3 command. I uninstalled them both and downloaded the zip version from this repository, then installed according to the instructions after running `make` command. Now `offlineimap` works correctly. 

Thank you for helping!

**Comment by nicolas33** [comment]
You might like to install offlineimap from the deb, though.

Thanks for the feedback.

**Comment by albins** [comment]
For the record I got this error when installing from homebrew on macOS, the equivalent of installing it from the deb, and just gave up and used mbsync instead. It seems that whatever the packagers are doing it's not enough to ensure isolation, or something's changed in Python.

**Comment by the-x-at** [comment]
Indeed, I am getting the same error message for `offlineimap` 7.3.4 under Python3.7. More precisely, I am on Mac OSX 10.12, Python 3.7.12 (installed via Fink). I migrated from Python2.7 today, where I was using flawlessly `offlineimap` 7.3.0. So I went to back to Python2.7, upgraded `offlineimap` to 7.3.4 and all works fine (again). Both installations were carried out using `pip` (also from the Fink distribution).
